web/src/main/java/org/cbioportal/web/SessionServiceController.java

```diff
@@ -117,7 +130,13 @@ private ResponseEntity<Session> addSession(SessionType type, Optional<SessionOpe
                 if (!(isAuthorized())) {
                     return new ResponseEntity<>(HttpStatus.SERVICE_UNAVAILABLE);
                 }
-                StudyPageSettings studyPageSettings = mapper.readValue(body.toString(), StudyPageSettings.class);
+                Class<? extends PageSettingsData> pageDataClass = pageTypeToData.get(
+                    SessionPage.valueOf((String) body.get("page"))
+                );
+                PageSettingsData studyPageSettings = mapper.readValue(
```

Comment:
If I understand it right, we are extending the use of page settings from study view page to results view. So maybe we should rename this to `pageSettings`
```suggestion
                PageSettingsData pageSettings = mapper.readValue(
```